### PR TITLE
Add Java specific modifiers, change Java symbol implementation to only include explicit modifiers

### PR DESCRIPTION
**Pull request by neetopia**

### CODE REVIEW

**Comment by neetopia** [comment]
#33

File: libraries/tools/kotlin-symbol-processing-api/src/org/jetbrains/kotlin/ksp/symbol/Modifier.kt [hunk 1]
```diff
@@ -17,5 +17,6 @@ enum class Modifier {
     ABSTRACT, FINAL, OPEN,
     VARARG, NOINLINE, CROSSINLINE,
     REIFIED,
-    EXPECT, ACTUAL
+    EXPECT, ACTUAL,
+    JAVA_DEFAULT, JAVA_NATIVE, JAVA_STATIC, JAVA_STRICT, JAVA_SYNCHRONIZED, JAVA_TRANSIENT, JAVA_VOLATILE
```
Comment (ting-yuan) [inline libraries/tools/kotlin-symbol-processing-api/src/org/jetbrains/kotlin/ksp/symbol/Modifier.kt:25]: May be worth commenting that
1. Some modifiers are shared but may have different semantics among languages.
2. Only explicit modifiers are observable. Therefore, use helpers like `isOpen` to check accessibility when possible.

File: plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/java/KSFunctionDeclarationJavaImpl.kt [hunk 2]
```diff
@@ -41,7 +41,12 @@ class KSFunctionDeclarationJavaImpl(val psi: PsiMethod) : KSFunctionDeclaration
     override val functionKind: FunctionKind = if (psi.hasModifier(JvmModifier.STATIC)) FunctionKind.STATIC else FunctionKind.MEMBER
 
     override val modifiers: Set<Modifier> by lazy {
-        psi.toKSModifiers()
+        val modifiers = mutableSetOf<Modifier>()
+        modifiers.addAll(psi.toKSModifiers())
+        if (psi.modifierList.hasExplicitModifier("default")) {
+            modifiers.add(Modifier.JAVA_DEFAULT)
+        }
```
Comment (ting-yuan) [inline plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/java/KSFunctionDeclarationJavaImpl.kt]: Why not add this into `toKSModifiers()`?

One related issue: in `toKSModifiers()`, would the map be created each time it is called?

File: plugins/ksp/src/org/jetbrains/kotlin/ksp/processor/JavaModifierProcessor.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            classDeclaration.declarations.map { it.accept(this, data) }
+        }
+
+        override fun visitPropertyDeclaration(property: KSPropertyDeclaration, data: Unit) {
+            this.visitModifierListOwner(property, data)
+        }
+
+        override fun visitFunctionDeclaration(function: KSFunctionDeclaration, data: Unit) {
+            this.visitModifierListOwner(function, data)
+        }
```
Comment (ting-yuan) [inline plugins/ksp/src/org/jetbrains/kotlin/ksp/processor/JavaModifierProcessor.kt:39]: These 3 overrides (`visitClassDeclaration`, `visitPropertyDeclaration`, `visitFunctionDeclaration`) seem unnecessary.

Comment (neetopia) [reply plugins/ksp/src/org/jetbrains/kotlin/ksp/processor/JavaModifierProcessor.kt:39]: one example is getQualifier is not implemented for Java symbols yet, so overriding them can avoid running into not implemented operations in TopDown visitor.

File: plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/java/KSFunctionDeclarationJavaImpl.kt (hunk 2 above)
Comment (neetopia) [reply plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/java/KSFunctionDeclarationJavaImpl.kt]: "default" is only valid on function declaration. but moving into to KSModofiers() will also work.

**Review by ting-yuan — APPROVED** [review]

**Pull request merged**